src/main/java/dbaccessor/user/UserAccessor.java

```diff
@@ -30,17 +30,19 @@ public long insert(User user) {
     }
 
     /* Checks if a user with the given username or email-id already exists in User table */
-    public boolean checkIfUserExists(String username, String emailID) {
+    public User checkIfUserExists(String username, String emailID) {
         String SQLStatment = "SELECT Username FROM User WHERE Username=@Username OR EmailID=@EmailID";
         Statement statement = Statement.newBuilder(SQLStatment)
                                 .bind("Username")
                                 .to(username)
                                 .bind("EmailID")
                                 .to(emailID)
                                 .build();
-        return !spannerTemplate
-                .query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true)) //setAllowPartialRead for reading specific columns
-                .isEmpty();
+        List<User> resultSet = spannerTemplate.query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true));
+        if(resultSet.isEmpty()) {
+            return null;
+        }
+        return resultSet.get(0);
```

Comment:
This function is being used at 2 places-
1. One inside the native signup API, so for that case it won't matter which user is returned (because it just checks if user is not null the throw `UserAlreadyExisitsException`).
2. Second, in `getUser` function of `GoogleAuthenticator` class. Here we want that only if there exists a unique entry in User table with both the username and email matching then that user should be returned.
I will update this function accordingly.

Reply:
Done.